src/path.c

```diff
@@ -1,50 +1,43 @@
 #include "path.h"
 
+enum path_state {
+	ORTHOGONAL, /**< When coming from an orthogonal movement */
+	DIAGONAL,   /**< When coming from diagonal movement */
+	PATH_STATES_COUNT,
+};
+
 struct translation {
 	char *from;
 	enum movement to;
 };
 
-struct dictionary {
-	struct translation *from_straight;
-	struct translation *from_diagonal;
-};
-
 // clang-format off
 /**
- * @brief These languages define ways to translate to a smooth path.
- *
- * Languages:
- *
- * - PATH_SAFE does not use diagonals
- * - PATH_DIAGONALS uses diagonals
- *
- * Each language defines two translation tables. One that is used when in a
- * straight movement and another that is used when in a diagonal movement.
+ * @brief This dictionary defines different ways to translate to a smooth path.
  */
-const struct dictionary languages[LANGUAGES_COUNT] = {
+static struct translation *dictionary[LANGUAGES_COUNT][PATH_STATES_COUNT] = {
```

Comment:
Just to say something, (it is also ok for me like this), instead the name dictionary, maybe something more particular?